### ISSUE TITLE
Radially averaged power spectrum

### ISSUE THREAD

**Issue by leouieda** [fatiando/fatiando#224]
Craig Miller [suggested on the maling list](https://groups.google.com/d/topic/fatiando/ULH80_AUrko/discussion) that we have a function to calculate the radially averaged power spectrum of grid data:

> Would it be possible to include a routine to calculate the radially averaged power spectrum of a grid. I currently read in the grid with the gridder.load_surfer and then i do some work using scipy to calculate the fft etc but it seems like it would be a useful thing to have as a routine for regular use.  I'm finding it tricky to get the proper wavenumber coordinates on the output (ie related to the original input grid).  It would be nice to have a suit of commonly used signal processing tools such as this included in the package.

This has applications mainly in potential field processing. A widely used application (too widely, if you ask me) is to separate shallow and deep sources [(Spector and Grant, 1970)](http://dx.doi.org/10.1190/1.1440092). This doesn't really work in most cases but the power spectrum is used elsewhere so it would be useful to have anyway.

Examples from the e-mail:

![](https://lh3.googleusercontent.com/-D0t3l8afcKI/VcKejZUyx9I/AAAAAAAAo2w/-_TxzBMYR98/s320/Boug_Anom_2300.png) ![](https://lh3.googleusercontent.com/-67TbGyChUN4/VcKelsi8iFI/AAAAAAAAo24/aKH5jM0yssM/s320/Boug_Anom_2300_fft_psd2d.png) ![](https://lh3.googleusercontent.com/-n0yED5K-Dj0/VcKenuL6A6I/AAAAAAAAo3A/aYsC-BGsQEs/s320/Boug_Anom_2300_fft_psd1d.png)

Getting the proper wavenumbers is really tricky because assumptions on the coordinate systems of the grid (x is North in Fatiando but sometimes people use x as East). Also, you can get tripped up by `numpy.meshgrid` if you pass and receive things in the wrong order.

Functions in `fatiando.gravmag.transform` calculate wavenumbers in a more reliable and tested way. Using them also reduced code duplication. The wavenumber functions don't really have a good interface as they were originally meant for internal use only. They should be given better names and docstrings and be cleaned up a bit.

**Comment by eusoubrasileiro** [comment]
Nice, seams useful. Wondering how to estimate a 1D power spectrum of wave number for a scatter set.

**Comment by leouieda** [comment]
Easiest way is to grid and FFT. But a more complicated way could be to use an equivalent layer (note: this is wild speculation). You could fit the layer in the space domain but then forward model in the frequency domain (the layer is a regular grid so FFT is OK).

**Comment by leouieda** [comment]
Actually, that might be a very cool idea for other things as well. (cc @birocoles)

**Comment by santis19** [comment]
In my Degree Thesis I've written some simple interactive software to estimate Curie's Depth using the power spectrum of magnetic anomalies. I've implemented some raw code to achive the radial average. I'll try to polish it to a more general application and upload it.

In my case, I forced the software to use square grids with even number of points on each axis. I've done this in order to get a grid point on its center, which is associated with the k=0 in the frequency domain.

I'll upload a simple code to start working on it.
